kernel/kernel-api/src/main/java/io/delta/kernel/internal/snapshot/LogSegment.java

```diff
@@ -217,6 +221,35 @@ public Optional<Long> getCheckpointVersionOpt() {
     return checkpointVersionOpt;
   }
 
+  /**
+   * Returns the most recent checksum file encountered during log file listing, if available.
```

Comment:
Is one of the guarantees that the checksum verison here is <= the `version` of this `LogSegment`?

Can we add a `checkArgument` for that?

Reply:
And is it ever useful to have a .crc file that is lower than the latest complete checkpoint of this logsegment?

i.e. are we able to define more strict bounds here?

Reply:
Done

>And is it ever useful to have a .crc file that is lower than the latest complete checkpoint of this logsegment?

One interesting thing is file size histogram, but I am not aware of the current use case needed to load it from previous CRC now or in the short term. We could make logSegment.getLastSeenChecksum() only return crc older than checkpoint for now as It is not a one way door.